src/toil/fileStore.py

```diff
@@ -64,10 +70,337 @@ def __init__(self, jobStore, jobWrapper, localTempDir, inputBlockFn):
+
+        If a user path is specified, it is used as the destination. If a user path isn't
+        specified, the file is stored in the local temp directory with an encoded name.
+
+        :param FileID fileStoreID: job store id for the file
+        :param string userPath: a path to the name of file to which the global file will be copied
+               or hard-linked (see below).
+        :param boolean cache: Described in :func:`~filestore.Filestore.readGlobalFile`
+        :param boolean mutable: Described in :func:`~filestore.Filestore.readGlobalFile`
+        :returns: An absolute path to a local, temporary copy of the file keyed by fileStoreID.
+        :rtype: string
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def readGlobalFileStream(self, fileStoreID):
+        """
+        Similar to readGlobalFile, but allows a stream to be read from the job store.
+        The yielded file handle does not need to and should not be closed explicitly.
+
+        :returns: a context manager yielding a file handle which can be read from.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def deleteLocalFile(self, fileStoreID):
+        '''
+        Deletes Local copies of files associated with the provided job store ID.
+
+        :param str fileStoreID: File Store ID of the file to be deleted.
+        :returns: None
+        '''
+        raise NotImplementedError()
+
+    @abstractmethod
+    def deleteGlobalFile(self, fileStoreID):
+        """
+        Deletes local files with the provided job store ID and then permanently deletes them from
+        the job store.
+        To ensure that the job can be restarted if necessary, the delete will not happen until
+        after the job's run method has completed.
+
+        :param fileStoreID: the job store ID of the file to be deleted.
+        :returns: None
+        """
+        raise NotImplementedError()
+
+    # Functions used to read and write files directly between a source url and the job store.
+    def importFile(self, srcUrl, sharedFileName=None):
+        return self.jobStore.importFile(srcUrl, sharedFileName=sharedFileName)
+
+    def exportFile(self, jobStoreFileID, dstUrl):
+        self.jobStore.exportFile(jobStoreFileID, dstUrl)
+
+    # A utility method for accessing filenames
+    def _abspath(self, key):
+        """
+        Return the absolute path to key.  This is a wrapper for os.path.abspath because mac OS
+        symlinks /tmp and /var (the most common places for a default tempdir) to /private/tmp and
+        /private/var respectively.
+
+        :param str key: The absolute or relative path to the file. If relative, it must be
+        relative to the local temp working dir
+        :returns: Absolute path to key
+        :rtype: str
+        """
+        if key.startswith('/'):
+            return os.path.abspath(key)
+        else:
+            return os.path.join(self.localTempDir, key)
```

Comment:
If the point of this method is to resolve symlinks, then this branch in the control flow doesn't achieve that. 

In /a/b/c, any component thereof could be a symlink. So /a/b could be a symlink, while c isn't.